### ISSUE TITLE
`Contents` scroll-to-top functionality should have pointer-cursor on hover

### ISSUE THREAD

**Issue by craiglabenz** [flutter/website#9189]
### Describe the problem

When a user scrolls down on the page, clicking `Contents` on the right hand side will scroll them back to the top of the page. However, to indicate this functionality, the cursor should be changed to a pointer for that element.

The trick is to only do this when the user has scrolled down some, otherwise clicking that element does nothing and thus the pointer cursor would feel broken.
